packages/vite/src/node/plugins/importAnalysisBuild.ts

```diff
@@ -118,7 +118,8 @@ export function buildImportAnalysisPlugin(config: ResolvedConfig): Plugin {
     async transform(source, importer) {
       if (
         importer.includes('node_modules') &&
-        !source.includes('import.meta.glob')
+        !source.includes('import.meta.glob') &&
+        !/import\s*\(/.test(source)
```

Comment:
I think we could extract this regex to a constant to avoid recreating it each time, apart from that LGTM 👍🏼